### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by moneromooo-monero**

### CODE REVIEW

**Comment by moneromooo-monero** [comment]
DO NOT MERGE YET

**Comment by moneromooo-monero** [comment]
Aaaaaand I uploaded an old version. Fixing...

**Comment by moneromooo-monero** [comment]
STILL DO NOT MERGE YET

**Comment by NanoAkron** [comment]
So there's an interesting philosophical point here - we use an adversarial system to achieve consensus on the network transaction record to prevent censorship. Yet here a three letter agency need only attack the single source of updates to gain control. 

Is there a way we can safely distribute our software in a censorship resistant manner?

**Comment by moneromooo-monero** [comment]
It already does. pigeons will work on reproducible builds. As for the rest, we'd need to make sure it's hard to sybil. This all sounds a bit like "wouldn't it be nice if we all had flying cars". Sure, but you might notice the reason it's not there yet isn't that nobody realized it was a good idea (and I'm not sure about the flying cars one).

So, maybe, but not now.

**Comment by moneromooo-monero** [comment]
Update to match URLs being less canonical than I'd want them.

STILL DO NOT MERGE

**Comment by fluffypony** [comment]
@NanoAkron they don't need to just compromise the website. They'd also have to compromise the hashes, which are GPG signed by me (on the website) and also in a DNSSEC-signed record (which is what the updater uses). Compromising my GPG key wouldn't compromise the website or the DNSSEC-signed record. Compromising the DNSSEC-signed record wouldn't compromise the website or my GPG key.

It would be nice if someone built a tool / hosted a site (eventually) that checked if the GPG-signed hashes correctly use my signature, and if the hashes match those in the DNSSEC-signed records.

File: src/cryptonote_core/cryptonote_core.cpp [hunk 1]
```diff
@@ -682,6 +706,18 @@ namespace cryptonote
     return true;
   }
   //-----------------------------------------------------------------------------------------------
+  bool core::check_tx_inputs_keyimages_domain(const transaction& tx) const
+  {
+    std::unordered_set<crypto::key_image> ki;
+    for(const auto& in: tx.vin)
+    {
+      CHECKED_GET_SPECIFIC_VARIANT(in, const txin_to_key, tokey_in, false);
+      if (!(rct::scalarmultKey(rct::ki2rct(tokey_in.k_image), rct::curveOrder()) == rct::identity()))
```
Comment (daira) [inline src/cryptonote_core/cryptonote_core.cpp]: 👍

**Comment by iridiumblue** [comment]
Wow - this is what a billion-dollar act of integrity looks like.

Respect.

**Comment by nfd9001** [comment]
Badass! Thanks.

**Comment by stefek99** [comment]
I do not fully understand the mechanism and / or implications.

Arrived here via blog some time ago: https://getmonero.org/2017/05/17/disclosure-of-a-major-bug-in-cryptonote-based-currencies.html

Now I'm looking at https://cryptonotestarter.org/inner.html and https://github.com/cryptonotefoundation/cryptonote

> Latest commit 8edd998  on Jan 18, 2016

I should probably notify them and request update.

**Pull request merged**